compiler/src/bytecode.rs

```diff
@@ -21,120 +28,213 @@ impl Bytecode {
         self.len() == 0
     }
 
+    /// emits a signed 64 bits integer
+    pub fn emit_int(&mut self, value: i64) {
+        self.bytes.extend(value.to_be_bytes());
+    }
+
+    /// emits an unsigned byte
+    pub fn emit_byte(&mut self, value: u8) {
+        self.bytes.push(value);
+    }
+
+    /// emits an unsigned 32 bits integer
+    pub fn emit_u32(&mut self, value: u32) {
+        self.bytes.extend(value.to_be_bytes());
+    }
+
+    /// emits a signed 64 bits float
+    pub fn emit_float(&mut self, value: f64) {
+        self.bytes.extend(value.to_be_bytes());
+    }
+
+    /// emits a constant pool reference, which is an unsigned 32 bits integer
+    pub fn emit_constant_ref(&mut self, constant: u32) {
+        self.emit_u32(constant);
+    }
+
+    /// Fills an instruction pointer at given instruction pointer in the byte array
+    pub fn patch_u32_placeholder(&mut self, placeholder: Placeholder, value: u32) {
```

Comment:
placeholders can be of any length, even if the `Bytecode` impl only provide u32 placeholders i wanted to explicit the type of placeholder

Reply:
This is still false to call it a `u32_placeholder` as it is a four-byte placeholder.

Reply:
it's not just a "four byte" placeholder, it's a `u32` placeholder. For now we only have one type of placeholded value, but i still want to underline the fact that the placeholder is an `u32`.

Reply:
The bytecode holds bytes, nothing more. The `u32` type is a representation for something that can be inserted in the byte vector, but does not convey any meaning. The type is also present in the method signature, this kinda repeat the same information. I would consider that placeholders are something that are only meaningful in the struct itself, and are re-exposed with more context than the internal type in new public methods. This is an address placeholder, so I would call it `patch_adress()`.

Reply:
its a _u32_ placeholder, we want to encapsulate the emission and patch of placeholders and not just consider them as "bytes holes" because it is the meaning of the `Bytecode` structure. `emit_u32_placeholder` comes with `patch_u32_placeholder`. I want to explicit that we manipulate `u32` placeholders because as you said placeholders are just an area wating for some undetermined bytes.